packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart

```diff
@@ -552,59 +552,49 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface
     String? password,
     SignInOptions? options,
   }) async {
-    bool isSignedIn;
-    try {
-      isSignedIn = (await fetchAuthSession()).isSignedIn;
-    } on Exception {
-      isSignedIn = false;
-    }
-    if (isSignedIn) {
-      throw const InvalidStateException(
-        'A user is already signed in.',
-        recoverySuggestion:
-            'Sign out the current user by calling `Amplify.Auth.signOut` and try the sign in again.',
-      );
-    }
-
     final pluginOptions = reifyPluginOptions(
       pluginOptions: options?.pluginOptions,
       defaultPluginOptions: const CognitoSignInPluginOptions(),
     );
 
-    // Create a new state machine for every call since it caches values
-    // internally on each run.
-    final stream = _stateMachine.create(SignInStateMachine.type).stream;
-    await _stateMachine
-        .accept(
-          SignInEvent.initiate(
-            authFlowType: pluginOptions.authFlowType,
-            parameters: SignInParameters(
-              (p) => p
-                ..username = username
-                ..password = password,
-            ),
-            clientMetadata: pluginOptions.clientMetadata,
+    // Cancel the current sign in, if any.
+    await _stateMachine.acceptAndComplete<SignInNotStarted>(
+      const SignInEvent.cancelled(),
+    );
```

Comment:
`SignInStateMachine.resolve()` calls `_assertSignedOut()` and `_reset()` before running the `SignInEvent.initiate`. why do we need to cancel sign in flow if any, isn't it taken care by the SignInStateMachine?

Reply:
Hmm yes this was leftover from earlier iterations. Removed 👍